mono/mini/driver.c

```diff
@@ -2231,7 +2231,20 @@ mono_main (int argc, char* argv[])
 		return 1;
 	}
 
-#if !defined(HOST_WIN32) && defined(HAVE_UNISTD_H)
+/*
+ * XXX: verify if other OSes need it; many platforms seem to have it so that
+ * mono_w32process_get_path -> mono_w32process_get_name, and the name is not
+ * necessarily a path instead of just the program name
+ */
+#if defined (_AIX)
```

Comment:
On Solaris `/proc/%d/path/a.out` returns full path. AIX probably uses`/proc/%d/object/a.out` instead (like old Solaris versions).

Reply:
they do have the file itself, but not its path, and it's not any kind of link either:

```
aix$ ls -l /proc/7733404/object/a.out
-rwxr-xr-x 1 root system 810463 Jun  5 08:05 /proc/7733404/object/a.out
aix$ file /proc/7733404/object/a.out
/proc/7733404/object/a.out: executable (RISC System/6000) or object module not stripped
```

and Solaris, for good measure:

```
calvin@hako:~$ ls -l /proc/7829/object/a.out
-r-xr-xr-x   1 root     bin      3236820 Mar 24 09:16 /proc/7829/object/a.out
calvin@hako:~$ file /proc/7829/object/a.out
/proc/7829/object/a.out:        ELF 32-bit LSB executable 80386 Version 1 [FPU], dynamically linked, not stripped
```

(FWIW, on AIX, the internal kernel structures for processes keep track of inode, for presumably what procfs is doing, but it doesn't expose that in userland APIs beyond what procfs is doing, though kernel modules can peer into it....)

granted, you could still launch from that executable in procfs, but it'd seem dirty IMHO; if I did do that though, i'll change this def to PASE and OpenBSD (plus maybe Haiku/Linux/non-x86 Darwin, since those backends do the same thing for path/name?)

Reply:
Our build is quite broken on Solaris, I tried recently..

Reply:
if I can get this admin of the Solaris system I'm using to install Sun Studio (and/or GCC) whenever he gets the time, I'll see what I can do WRT that....

Reply:
It is easy to try Solaris/x86 and Solaris/amd64 in VMware, etc.
I have access to SPARC systems, but..it isn't a priority.

Reply:
https://github.com/mono/mono/issues/6949

Reply:
looks like Linux has congruent path == name where name is equiv to the full path, but only if the first two attempts at reading procfs succeed, otherwise you just get the name (not 

so in that case, the OSes that would suffer without this are AIX/PASE, OpenBSD, Solaris, Haiku, and non-x86 Darwin; and I've only tested AIX and fixed FreeBSD (as well as NetBSD), so... (i'll consider testing Solaris whenever the guy gets back to me; Haiku depends on me getting it in a less broken state, etc.; so the others can be added when/if the need arises for them, especially if proper path fetching could be added)